cornice/validators/__init__.py

```diff
@@ -33,7 +34,8 @@ def extract_cstruct(request):
     if request.content_type == 'application/x-www-form-urlencoded':
         body = request.POST.mixed()
 
-    elif request.content_type and request.content_type != 'application/json':
+    elif (request.content_type and
+          not re.match('^application/(.*?)json$', str(request.content_type))):
```

Comment:
Side note, Pyramid has compat module to avoid using `six`:
http://docs.pylonsproject.org/projects/pyramid/en/latest/api/compat.html